tests/clients/test_psycopg2.py

```diff
@@ -48,341 +50,366 @@ def setUp(self):
         cursor = self.db.cursor()
         cursor.execute(database_setup_query)
         self.db.commit()
-        cursor.close()
-
 
         self.cursor = self.db.cursor()
-        self.recorder = tracer.recorder
+        self.recorder = tracer.span_processor
         self.recorder.clear_spans()
         tracer.cur_ctx = None
-
-    def tearDown(self):
+        yield
         if self.cursor and not self.cursor.connection.closed:
-          self.cursor.close()
+            self.cursor.close()
         if self.db and not self.db.closed:
-          self.db.close()
+            self.db.close()
         agent.options.allow_exit_as_root = False
 
-    def test_vanilla_query(self):
-        self.assertTrue(psycopg2.extras.register_uuid(None, self.db))
-        self.assertTrue(psycopg2.extras.register_uuid(None, self.db.cursor()))
+    def test_register_json(self) -> None:
+        resp = register_json_with_instana(conn_or_curs=self.db)
+        assert resp[0].values[0] == 114
+        assert resp[1].values[0] == 199
+
+    def test_vanilla_query(self) -> None:
+        assert psycopg2.extras.register_uuid(None, self.db)
+        assert psycopg2.extras.register_uuid(None, self.db.cursor())
 
         self.cursor.execute("""SELECT * from users""")
         affected_rows = self.cursor.rowcount
-        self.assertEqual(1, affected_rows)
+        assert 1 == affected_rows
         result = self.cursor.fetchone()
 
-        self.assertEqual(6, len(result))
+        assert 6 == len(result)
 
         spans = self.recorder.queued_spans()
-        self.assertEqual(0, len(spans))
+        assert 0 == len(spans)
 
-    def test_basic_query(self):
-        with tracer.start_active_span('test'):
+    def test_basic_query(self) -> None:
+        with tracer.start_as_current_span("test"):
             self.cursor.execute("""SELECT * from users""")
             affected_rows = self.cursor.rowcount
             result = self.cursor.fetchone()
             self.db.commit()
 
-        self.assertEqual(1, affected_rows)
-        self.assertEqual(6, len(result))
+        assert 1 == affected_rows
+        assert 6 == len(result)
 
         spans = self.recorder.queued_spans()
-        self.assertEqual(2, len(spans))
+        assert 2 == len(spans)
 
         db_span, test_span = spans
 
-        self.assertEqual("test", test_span.data["sdk"]["name"])
-        self.assertEqual(test_span.t, db_span.t)
-        self.assertEqual(db_span.p, test_span.s)
+        assert "test" == test_span.data["sdk"]["name"]
+        assert test_span.t == db_span.t
+        assert db_span.p == test_span.s
 
-        self.assertIsNone(db_span.ec)
+        assert db_span.ec is None
```

Comment:
You may have to make similar changes at multiple places in this file.

```suggestion
        assert not db_span.ec
```

Reply:
Done